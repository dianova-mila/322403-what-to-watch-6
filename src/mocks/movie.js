export default {
  name: `Macbeth`,
  posterImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  rating: 3.3,
  scoresCount: 48798,
  director: `Justin Kurzel`,
  starring: [
    `Michael Fassbender`,
    `Marion Cotillard`,
    `Jack Madigan`
  ],
  runTime: 113,
  genre: `Drama`,
  released: 2015,
  id: `1`,
  isFavorite: false,
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
};
