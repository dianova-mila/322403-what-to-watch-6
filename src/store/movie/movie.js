import {createReducer} from "@reduxjs/toolkit";
import {loadOneMovie, loadPromoMovie, loadComments} from "../actions";

const initialState = {
  movie: {},
  promoMovie: {},
  isPromoMovieLoaded: false,
  comments: [],
  isOneMovieLoaded: false,
};

const movie = createReducer(initialState, (builder) => {
  builder.addCase(loadOneMovie, (state, action) => {
    state.movie = action.payload;
    state.isOneMovieLoaded = true;
  });

  builder.addCase(loadPromoMovie, (state, action) => {
    state.promoMovie = action.payload;
    state.isPromoMovieLoaded = true;
  });

  builder.addCase(loadComments, (state, action) => {
    state.comments = action.payload;
  });
});

export {movie};
