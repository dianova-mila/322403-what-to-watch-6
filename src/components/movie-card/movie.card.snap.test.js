import React from "react";
import {Router} from "react-router-dom";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {createMemoryHistory} from "history";
import movie from "../../mocks/movie";
import MovieCard from "./movie-card";

const authorizationStatus = true;
const userInfo = {avatarUrl: `img/1.png`};

const onUserAvatarClick = jest.fn();
const onPlayButtonClick = jest.fn();
const mockStore = configureStore({});


it(`MovieCard should render correctly`, () => {
  const history = createMemoryHistory();

  const {container} = render(
      <Provider store={mockStore({
        USER: {authorizationStatus, userInfo},
        MOVIE: {promoMovie: movie, isPromoMovieLoaded: true}
      })}>
        <Router history={history}>
          <MovieCard onUserAvatarClick={onUserAvatarClick} onPlayButtonClick={onPlayButtonClick}/>
        </Router>
      </Provider>
  );

  expect(container).toMatchSnapshot();
});
