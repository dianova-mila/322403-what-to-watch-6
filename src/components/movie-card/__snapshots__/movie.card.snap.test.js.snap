// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img
        alt="Macbeth"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg"
      />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/1.png"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            alt="Macbeth"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            Macbeth
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Drama
            </span>
            <span
              class="movie-card__year"
            />
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
